config/initializers/controllers.rb

```diff
@@ -77,13 +80,3 @@ def last_signin_provider
   end
 
 end
-
-# Layout is not inheritable in Rails 3.2
-Rails.application.config.to_prepare do
-  # Put this code inside the class_exec above in Rails 4
-  # and remove it from ApplicationController
-  [Doorkeeper::ApplicationController,
-   FinePrint::ApplicationController].each do |klass|
-    klass.layout 'application'
-  end
-end
```

Comment:
Doorkeeper as usual uses its own completely separate layout... FinePrint seems to work.